### ISSUE TITLE
-mapfields-as-maps and -json possible incompability

### ISSUE THREAD

**Issue by andreeelopes** [tomas-abrahamsson/gpb#227]
Hey, it's possible I missed something in the docs regarding this subject, but there seems to be an issue with the json decoding when `-mapfields-as-maps` is passed to `protoc-erl`.

```proto
syntax = "proto3";

message Person {
  string name = 1;
  map<string, string> map = 2;
}
```

```shell
../gpb/bin/protoc-erl -I. -mapfields-as-maps -json  x.proto

erlc -I../gpb/include x.erl
```
   

```erlang
Erlang/OTP 24 [erts-12.3.2.2] [source] [64-bit] [smp:10:10] [ds:10:10:10] [async-threads:1]

Eshell V12.3.2.2  (abort with ^G)
1> rr("x.hrl").
['Person']
3>
3> Bin = x:encode_msg(#'Person'{name="abc def", map=#{"a"=>"b"}}).
<<10,7,97,98,99,32,100,101,102,18,6,10,1,97,18,1,98>>
4>
4> Record = x:decode_msg(Bin, 'Person').
#'Person'{name = "abc def", map = #{"a" => "b"}}
5>
5> Map = x:to_json(Record).
** exception error: no match of right hand side value #{key => "a",
                                                        value => "b"}
     in function  x:'-tj_mapfield_fold/3-fun-0-'/4 (x.erl, line 624)
     in call from lists:foldl/3 (lists.erl, line 1267)
```

Could we add an extra condition [here](https://github.com/tomas-abrahamsson/gpb/blob/5557e3117235b7df70a302c4f18267ec95777803/src/gpb_gen_json_encoders.erl#L1041) when `records` is matched, to also check if `mapfields-as-maps` is on?

Thanks in advance!

**Comment by tomas-abrahamsson** [comment]
Hi, you're absolutely right, that option was not properly handled together with the json option. Thanks for making me aware. I believe it is now fixed in the 4.19.8 release, which is fresh from the press. I'll close this issue. If you should find there's anything more to it, feel free to just reopen, or open another issue.

**Comment by tomas-abrahamsson** [comment]
Closing.
